### PR TITLE
Display coverage status in PRs

### LINKED ISSUE #2273: [Feature Request]: Display coverage status in PRs
### 🚀 The feature

anautsch has written up documentation on code coverage in some detail: https://github.com/speechbrain/speechbrain/blob/d71a7e018738329fe9a7951d2e9457d52f94edad/docs/coverage.md

The core team sees interest in providing code coverage statistics as part of the CI pipeline so that these can be displayed as part of pull requests.

### Solution outline

TODO, need to find a nice way to do it along our current CI pipeline.

### Additional context

~~I'm self-assigning for now, but I won't get to it right away. If anybody else wants to look into it, feel free to look into it and mention it here prior to starting work (to avoid duplicated work).~~

Pull request is under work, see linked PR.

### PR DESCRIPTION
**Pull request by asumagic**
## What does this PR do?

Leverages [`pytest-cov`](https://pypi.org/project/pytest-cov/) and the [`MishaKav/pytest-coverage-comment`](https://github.com/MishaKav/pytest-coverage-comment) GitHub action to collect coverage statistics and dump the information as a GitHub comment.

Currently WIP, making the PR so that I can actually test it.

The different test steps are merged into one in order to produce a single coverage file.

Fixes #2273

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by asumagic** [comment]
Ok, the permission error seems to be caused by the fact that `pull_request` triggers do not grant write access to the token, so the current approach in my PR seems to be incorrect.  
GitHub does this [for security reasons](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/), see https://github.com/actions/first-interaction/issues/10 (joining the club of cross-referencing that issue).

There is a `pull_request_target` trigger which *does* grant write access to the GitHub token. However, it executes with the workflow&code from in the main repository, not from the PR (which is untrusted). The linked blog post elaborates on a safe way to achieve this.

tl;dr @mravanelli: I'll keep on working on this issue but it needs a bit more thought. Nothing to change on your end.

**Comment by Adel-Moumen** [comment]
Hello @asumagic, do you have some news on this PR ? Thanks. 

Best, 
Adel

**Comment by asumagic** [comment]
Not at the moment, I need to change some of the approach due to the mentioned issues.

**Comment by Adel-Moumen** [comment]
We could use something like CodeCov: https://github.com/espnet/espnet/pull/5620#issuecomment-1885887920

Wdyt ?

**Comment by Adel-Moumen** [comment]
Another bot that could do the job : https://github.com/pylint-dev/pylint/pull/7398#issuecomment-1234740022

**Comment by Adel-Moumen** [comment]
Another thing in favour of codecov is that we can integrate it as part of our ci/cd tests. see lhost example: https://github.com/lhotse-speech/lhotse/pull/1031
